apps/passport-server/src/services/devconnect/organizerSync.ts

```diff
@@ -678,6 +730,28 @@ export class OrganizerSync {
               updatedTicket
             )}`
           );
+
+          // If a ticket has been checked in on Pretix, but was not checked
+          // in on our side, then consume it.
+          if (!oldTicket?.is_consumed && updatedTicket.is_consumed) {
+            updatedTicket.checker = "Pretix";
+          } else {
+            // Otherwise preserve whatever the checker value was from the
+            // old ticket
+            updatedTicket.checker = oldTicket?.checker as string;
+          }
+
+          // If the ticket was pending sync, ignore the fact that Pretix is
+          // telling us it's not checked in.
+          if (
+            oldTicket?.is_consumed &&
+            !oldTicket.pretix_checkin_timestamp &&
+            !updatedTicket.is_consumed
+          ) {
+            updatedTicket.checkin_timestamp = oldTicket.checkin_timestamp;
+            updatedTicket.is_consumed = true;
+          }
```

Comment:
this is a little confusing for me and seems to be at the heart of this pr. as the reader i want a little more hand-holding through this section.

Reply:
I've completely rewritten this section to be much clearer. I've also moved the checkin-related fields out of the DevconnectPretixTicket base interface, so that this interface now aligns solely with what comes back from Pretix or can be computed from it. This hopefully makes "where do the different pieces of data come from" much clearer.